app/lib/shared/search_service.g.dart

```diff
@@ -0,0 +1,87 @@
+// GENERATED CODE - DO NOT MODIFY BY HAND
+
+part of pub_dartlang_org.shared.search_service;
+
+// **************************************************************************
+// Generator: JsonSerializableGenerator
+// **************************************************************************
+
+PackageDocument _$PackageDocumentFromJson(Map<String, dynamic> json) =>
+    new PackageDocument(
+        url: json['url'] as String,
+        package: json['package'] as String,
+        version: json['version'] as String,
+        devVersion: json['devVersion'] as String,
+        description: json['description'] as String,
+        lastUpdated: json['lastUpdated'] as String,
+        readme: json['readme'] as String,
+        detectedTypes:
+            (json['detectedTypes'] as List)?.map((e) => e as String)?.toList(),
+        popularity: json['popularity'] as double);
```

Comment:
Side note: I'm not quite sure if the generator is doing this correctly: Often the wireformat json contains numbers as either as doubles or integers (if there is no fractional part). A `0 as double` will throw an exception!

So in order to work around this, it's better if the generator would generate: `(json['popularity'] as num).toDouble()` here. This ensures that the resulting value is a double, even if it was encoded as a whole integer.